src/comments.js

```diff
@@ -381,6 +383,20 @@ function handleConditionalExpressionComments(enclosingNode, followingNode, comme
   return false;
 }
 
+function handleAssignmentPatternComments(enclosingNode, comment) {
```

Comment:
@jlongster + @vjeux I think I've identified what's causing the issue. For example, with something like:

```js
const {a /* comment */ = 1 } = b
```

The comment is stripped and we get the error message for non-printed comments. This is coming from https://github.com/jlongster/prettier/blob/master/src/comments.js#L487 where the right-hand side of the boolean (`types.getFieldValue(value, "comments")`) is always `null` or `undefined`. As a consequence, on line 490, `printed` is returned and the comment is sadly omitted...